### PR TITLE
FS-4621 Add assessment assignment handlers

### PR DESCRIPTION
**Pull request by sfount**
### Change description
Use the latest version of the funding service shared utils to let us use
the new email template type enums for when assessments are assigned and
unassigned.

When the new template types for assessment assignment and unassignment
are provided, map the body contents to the new `Assignment` parser and
send to the new template ID provided in the default config.

- [x] Unit tests and other appropriate tests added or updated
- [x] README and other documentation has been updated / added (if needed)
- [x] Commit messages are meaningful and follow good commit message guidelines

### CODE REVIEW

File: app/notification/model/notification.py [hunk 1]
```diff
@@ -72,6 +72,18 @@ def email_recipient(json_data: dict):
                     template_name=NotifyConstants.TEMPLATE_TYPE_EOI_PASS_W_CAVEATS,
                 )
 
+            case NotifyConstants.TEMPLATE_TYPE_ASSESSMENT_APPLICATION_ASSIGNED:
```
Comment (nuwan-samarasinghe) [inline app/notification/model/notification.py:75]: Have we created the template in notification service (GOV Notify)

**Review by nuwan-samarasinghe — APPROVED** [review]

File: app/notification/model/notification.py (hunk 1 above)
Comment (sfount) [reply app/notification/model/notification.py:75]: I think that was done by others as part of the ticket, thanks for the prompt though we'll make sure its in -- I used some IDs I found in the ticket here https://github.com/communitiesuk/funding-service-design-notification/pull/258/files#diff-830b41638f32c4afb3ccef5bf4c2c26af86371a2e52b088f895297c5577bca63R92-R99

**Review by ksp37-dluhc — APPROVED** [review]

**Comment by ksp37-dluhc** [comment]
Just noticed but is this missing the `requirements-dev.in` updates ?

**Comment by sfount** [comment]
> Just noticed but is this missing the `requirements-dev.in` updates ?

Yeh I updated `reuirements.in` and ran `pip-compile requirements.in` and `pip-comile requirement-dev.in`, it looks like dev dependencies are "locked on top" and copy the underlying requirements lockfile.

I'm not too comfortable with it updating so many things at once but I suspect theres a backlog to get through

**Pull request merged**